### PR TITLE
 build/tools/amebalite/bootloader/: Update SPIC settings in bootlaoders and SDK, to fix the hanging/hard fault when XIP and SPIC user mode used.

### PR DESCRIPTION
**Pull request by fengyunlai**
1.Update SPIC settings in bootloaders and SDK, to fix hanging/hardfault issue with XIP.
2.Shortern the name of samsung reserved region's partition to not surpass 15.

### CODE REVIEW

File: build/configs/rtl8720e/hello/defconfig [hunk 1]
```diff
@@ -262,7 +262,7 @@ CONFIG_ARCH_BOARD_HAVE_FLASH=y
 CONFIG_FLASH_MINOR=0
 CONFIG_FLASH_PART_SIZE="32,28,16,4,432,3068,3068,512,1024,8,"
 CONFIG_FLASH_PART_TYPE="none,none,none,none,none,kernel,kernel,smartfs,none,bootparam,"
-CONFIG_FLASH_PART_NAME="bl,reserved,ftl,system_data,ss,kernel,kernel,userfs,samsung_reserved,bootparam,"
+CONFIG_FLASH_PART_NAME="bl,reserved,ftl,system_data,ss,kernel,kernel,userfs,ss_reserved,bootparam,"
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/hello/defconfig:265]: Could you let me know why you modify this?

File: os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_flashclk.c [hunk 2]
```diff
@@ -379,9 +509,10 @@ void flash_highspeed_setup(void)
...
 
 	if (SYSCFG_CHIPType_Get() != CHIP_TYPE_FPGA) {
-		flash_calibration_highspeed(flash_speed);
+		flash_handshake_highspeed(flash_speed);
 	}
 
 	__asm volatile("cpsie i");
 }
 
+s
```
Comment (sunghan-chang) [inline os/board/rtl8720e/src/component/soc/amebalite/fwlib/ram_km4/ameba_flashclk.c:518]: Please remove a typo here.

**Review by sunghan-chang — COMMENTED** [review]
Could you let me know what SPIC is?

File: build/configs/rtl8720e/hello/defconfig (hunk 1 above)
Comment (fengyunlai) [reply build/configs/rtl8720e/hello/defconfig:265]: If I enable CONFIG_MTD_PARTITION_NAMES, I will see "ERROR: Partition name is so long. Please make it smaller than 15". So I make it less than 15.

#ifdef CONFIG_MTD_PARTITION_NAMES
#define MTD_PARTNAME_LEN       15
#endif

**Comment by sunghan-chang** [comment]
@fengyunlai Could you resolve the conflicts?

**Comment by sunghan-chang** [comment]
Could you let me know what SPIC is

**Comment by sunghan-chang** [comment]
@fengyunlai Could you check my comments?

File: build/configs/rtl8720e/hello/defconfig [hunk 3]
```diff
@@ -262,7 +262,7 @@ CONFIG_ARCH_BOARD_HAVE_FLASH=y
 CONFIG_FLASH_MINOR=0
 CONFIG_FLASH_PART_SIZE="32,28,16,4,432,3068,3068,512,1024,8,"
 CONFIG_FLASH_PART_TYPE="none,none,none,none,none,kernel,kernel,smartfs,none,bootparam,"
-CONFIG_FLASH_PART_NAME="bl,reserved,ftl,system_data,ss,kernel,kernel,userfs,samsung_reserved,bootparam,"
```
Comment (sangam-swami) [inline build/configs/rtl8720e/hello/defconfig:265]: Spell mistake in commit message bootlaoders => bootloader

**Comment by fengyunlai** [comment]
> @fengyunlai Could you check my comments?

Sure Mr Chang, sorry for late reply. I was on vocation last 5 days. SPIC refers to SPI Flash controller. I will fix the conflict soon.

File: build/configs/rtl8720e/hello/defconfig (hunk 3 above)
Comment (fengyunlai) [reply build/configs/rtl8720e/hello/defconfig]: Corrected, thanks

**Review by sunghan-chang — APPROVED** [review]

**Pull request merged**